sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/azure_configuration_client_async.py

```diff
@@ -180,14 +195,44 @@ def list_configuration_settings(
                 key="MyKey", label="MyLabel"
             )
         """
-        error_map = {404: ResourceNotFoundError}
-        return await self._impl.get_configuration_setting(
+        if match_condition == MatchConditions.Unconditionally:
```

Comment:
I feel a little redundancy, I would have written:
``` python
error_map = {
       404: ResourceNotFoundError
}
if match_condition in [MatchConditions.IfNotModified, MatchConditions.IfPresent, MatchConditions.IfMissing]:
    error_map[412]=ResourceModifiedError
if match_condition in [MatchConditions.IfModified]:
    error_map[304]=ResourceNotModifiedError
```

Personally I find it easier to read and understand